data-prepper-plugins/saas-source-plugins/jira-source/src/test/java/org/opensearch/dataprepper/plugins/source/jira/JiraServiceTest.java

```diff
@@ -0,0 +1,315 @@
+        jiraSourceConfigMap.put("project", projectKey);
+
+
+        String jiraSourceConfigJsonString = objectMapper.writeValueAsString(jiraSourceConfigMap);
+        return objectMapper.readValue(jiraSourceConfigJsonString, JiraSourceConfig.class);
+    }
+
+    private IssueBean createIssueBean(boolean nullFields) {
+        IssueBean issue1 = new IssueBean();
+        issue1.setId(UUID.randomUUID().toString());
+        issue1.setKey("issue_1_key");
+        issue1.setSelf("https://example.com/rest/api/2/issue/123");
+        issue1.setExpand("operations,versionedRepresentations,editmeta");
+
+        Map<String, Object> fieldMap = new HashMap<>();
+        if (!nullFields) {
+            fieldMap.put(CREATED, Instant.now());
+            fieldMap.put(UPDATED, Instant.now());
+        } else {
+            fieldMap.put(CREATED, 0);
+            fieldMap.put(UPDATED, 0);
+        }
+
+        Map<String, Object> issueTypeMap = new HashMap<>();
+        issueTypeMap.put("name", "Task");
+        issueTypeMap.put("self", "https://example.com/rest/api/2/issuetype/1");
+        issueTypeMap.put("id", "1");
+        fieldMap.put("issuetype", issueTypeMap);
+
+        Map<String, Object> projectMap = new HashMap<>();
+        if (!nullFields) {
+            projectMap.put("name", "project name test");
+            projectMap.put(KEY, "TEST");
+        }
+        fieldMap.put("project", projectMap);
+
+        Map<String, Object> priorityMap = new HashMap<>();
+        priorityMap.put("name", "Medium");
+        fieldMap.put("priority", priorityMap);
+
+        Map<String, Object> statusMap = new HashMap<>();
+        statusMap.put("name", "In Progress");
+        fieldMap.put("status", statusMap);
+
+        issue1.setFields(fieldMap);
+
+        return issue1;
+    }
+
+    private void waitForFutures(List<Future<Boolean>> futureList) throws InterruptedException {
+        for (Future<?> future : futureList) {
+            try {
+                future.get();
+            } catch (InterruptedException ie) {
+                log.error("Thread interrupted.", ie);
+                throw new InterruptedException(ie.getMessage());
+            } catch (ExecutionException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"basic-auth-jira-pipeline.yaml"})
+    public void testFetchingJiraIssue(String configFileName) {
+        doReturn(new ResponseEntity<>("", HttpStatus.OK)).when(restTemplate).getForEntity(any(URI.class), any(Class.class));
+        JiraSourceConfig jiraSourceConfig = createJiraConfigurationFromYaml(configFileName);
+        JiraAuthConfig authConfig = new JiraAuthFactory(jiraSourceConfig).getObject();
+        JiraService jiraService = new JiraService(restTemplate, jiraSourceConfig, authConfig);
+        String ticketDetails = jiraService.getIssue("key");
+        assertNotNull(ticketDetails);
```

Comment:
Can you do more assertions here?

Reply:
Modified it to check equality with the expected ticket response